frameworklauncher/doc/USERMANUAL.md

```diff
@@ -633,24 +639,103 @@ Notes:
+    </tr>
+    <tr>
+      <td>TaskRole2</td>
+      <td><i>FancyRetryPolicy = false<br>MaxRetryCount = 0</i></td>
+    </tr>
+    <tr>
+      <td rowspan="1"><b>Service</td>
+      <td rowspan="1"><i>FancyRetryPolicy = false</i><br>MaxRetryCount = -2</td>
+      <td>TaskRole1</td>
+      <td><i>FancyRetryPolicy = false</i><br>MaxRetryCount = -2</td>
+      <td rowspan="1">Always Retry for any failure or success.</td>
+    </tr>
+    <tr>
+      <td rowspan="1"><b>Blind Batch Job</td>
+      <td rowspan="1"><i>FancyRetryPolicy = false</i><br>MaxRetryCount = -1</td>
+      <td>TaskRole1</td>
+      <td><i>FancyRetryPolicy = false</i><br>MaxRetryCount = -1</td>
+      <td rowspan="1">Always Retry for any failure.<br>Never Retry for success.</td>
+    </tr>
+    <tr>
+      <td rowspan="1"><b>Batch Job with Task Fault Tolerance</td>
+      <td rowspan="1">FancyRetryPolicy = true<br>MaxRetryCount = 3</td>
+      <td>TaskRole1</td>
+      <td>FancyRetryPolicy = true<br>MaxRetryCount = 3</td>
+      <td rowspan="1">Always Retry for transient failure.<br>Never Retry for non-transient failure or success.<br>Retry up to 3 times for unknown failure.</td>
+    </tr>
+    <tr>
+      <td rowspan="1"><b>Batch Job without Task Fault Tolerance</td>
+      <td rowspan="1">FancyRetryPolicy = true<br>MaxRetryCount = 3</td>
+      <td>TaskRole1</td>
+      <td><i>FancyRetryPolicy = false<br>MaxRetryCount = 0</i></td>
+      <td rowspan="1">For Framework RetryPolicy, same as "Batch Job with Task Fault Tolerance".<br>For Task RetryPolicy, because the Task cannot tolerate any failed Container, such as it cannot recover from previous failed Container, so Never Retry Task for any failure or success.</td>
+    </tr>
+    <tr>
+      <td rowspan="1"><b>Debug Mode</td>
+      <td rowspan="1">FancyRetryPolicy = true<br><i>MaxRetryCount = 0</i></td>
+      <td>TaskRole1</td>
+      <td>FancyRetryPolicy = true<br><i>MaxRetryCount = 0</i></td>
+      <td rowspan="1">Always Retry for transient failure.<br>Never Retry for non-transient failure or unknown failure or success.<br>This can help to capture the unexpected exit of User Service itself.</td>
+    </tr>
+  </tbody>
+</table>
+
+
 ## <a name="ApplicationCompletionPolicy">ApplicationCompletionPolicy</a>
 ### <a name="ApplicationCompletionPolicy_Overview">Overview</a>
 ApplicationCompletionPolicy can be configured for each TaskRole to control:
-1. The conditions to complete the Application.
-2. The ExitStatus of the completed Application.
+1. The conditions to complete the [Application](#Concepts_YARN).
+2. The ExitStatus of the completed [Application](#Concepts_YARN).
 
 ### <a name="ApplicationCompletionPolicy_Usage">Usage</a>
 For details, please check: [TaskRoleApplicationCompletionPolicyDescriptor](../src/main/java/com/microsoft/frameworklauncher/common/model/TaskRoleApplicationCompletionPolicyDescriptor.java).
 
 ### <a name="ApplicationCompletionPolicy_Examples">Examples</a>
```

Comment:
>Examples [](start = 51, length = 8)

it is weird you firstly introduce "MinFailedTaskCount" and "MinSucceededTaskCount" in an example. do you have anywhere else to explain the concept formally?

Reply:
They are in previous section [Usage] with hyperlink.

View it in the web is more friendly:
[https://github.com/Microsoft/pai/blob/yqwang/launcher-dev/frameworklauncher/doc/USERMANUAL.md#ApplicationCompletionPolicy](https://github.com/Microsoft/pai/blob/yqwang/launcher-dev/frameworklauncher/doc/USERMANUAL.md#ApplicationCompletionPolicy)


### <a name="ApplicationCompletionPolicy_Usage">Usage</a>
For details, please check: [TaskRoleApplicationCompletionPolicyDescriptor](../src/main/java/com/microsoft/frameworklauncher/common/model/TaskRoleApplicationCompletionPolicyDescriptor.java).

---
In reply to: [200238905](https://github.com/Microsoft/pai/pull/812#discussion_r200238905) [](ancestors = 200238905)

Reply:
I see. you introduce it in the comment of java code.

---
In reply to: [200240311](https://github.com/Microsoft/pai/pull/812#discussion_r200240311) [](ancestors = 200240311,200238905)